packages/jest-emotion/test/react-enzyme.test.js

```diff
@@ -1,60 +1,88 @@
 import 'test-utils/legacy-env'
 /** @jsx jsx */
+import cases from 'jest-in-case'
 import * as enzyme from 'enzyme'
 import { jsx } from '@emotion/core'
-import { createSerializer as createEnzymeSerializer } from 'enzyme-to-json'
 import { createSerializer } from 'jest-emotion'
-import { toMatchSnapshot } from 'jest-snapshot'
 import React from 'react'
-
-const createEnzymeSnapshotMatcher = serializerOptions => {
-  const serializer = createEnzymeSerializer(serializerOptions)
-  const identityPrinter = v => v
-
-  return function(val) {
-    return toMatchSnapshot.call(this, serializer.print(val, identityPrinter))
-  }
-}
+import toJson from 'enzyme-to-json'
 
 expect.addSnapshotSerializer(createSerializer())
 
-expect.extend({
-  toMatchShallowSnapshot: createEnzymeSnapshotMatcher(),
-  toMatchDeepSnapshot: createEnzymeSnapshotMatcher({ mode: 'deep' })
-})
+const testWithMethods = (description, testCase) =>
+  cases(description, testCase, [
+    { name: 'render component with shallow', method: 'shallow' },
+    { name: 'render component with mount', method: 'mount' }
+  ])
 
-test('enzyme mount test', () => {
+const expectToMatchSnapshot = (component, method, mode) => {
```

Comment:
I would move handling `method` into `testWithMethods`, and keep that helper focused on calling appropriate `mode`, that way the last test which is using this helper directly could just use method directly (on its own) and only delegate to this helper for handling mode - that alone test looks a little bit weird when it calls this method parametrized with both